### ISSUE TITLE
cannot run on gordon

### ISSUE THREAD

**Issue by haoyuanchen** [radical-cybertools/radical.repex.at#39]
Died in the preparation step, before generating running CUs.

**repex version**: 0.2-feature-tuu-opt5-d2eb6bc-
**rp version**: 0.35
fresh env, fresh install
**agent.err**:

intel/2013.1.117(49):ERROR:105: Unable to locate a modulefile for 'mkl'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^M 32 1957k   32  639k    0     0  1498k      0  0:00:01 --:--:--  0:00:01 3401k^M100 1957k  100 1957k    0     0  4104k      0 --:--:-- --:--:-- --:--:-- 8224k
no previously-included directories found matching 'libcloud/test/secrets.py'
Python 2.7.9
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/chen1990/.local/lib/python2.6/site-packages/saga_python-0.14-py2.6.egg/saga/**init**.py", line 47, in <module>
    short_version, long_version, branch = ru.get_version ([root, pwd])
ValueError: too many values to unpack

end of **agent.out**:

 (/home/chen1990/radical.pilot.sandbox/ve_gordon/bin/python)
PYTHONPATH: /home/chen1990/radical.pilot.sandbox/rp.session.prot-55-247.rutgers.edu.haoyuan.016693.0002-pilot.0000/rp_install/lib/python2.7/site-packages:::/home/chen1990/.local/lib/python2.6/site-packages
utils :  0.35 /home/chen1990/radical.pilot.sandbox/rp.session.prot-55-247.rutgers.edu.haoyuan.016693.0002-pilot.0000/rp_install/lib/python2.7/site-packages/radical/utils/**init**.pyc
saga  : 
install failed!
ESC[HESC[J

**Comment by andre-merzky** [comment]
can you please make sure that you also use 0.35 for saga-python and radical.utils?  Please let us know if the problem persists!

**Comment by haoyuanchen** [comment]
Hi Andre,

"sagapython-version" returns "0.35". For radical.utils I didn't see the version command but I think it should be 0.35 too because I uninstalled everything before I re-installed them and tried to run RepEx.

Thanks!

**Comment by andre-merzky** [comment]
Sorry, I should have seen this before, but it says right in your log:

```
/home/chen1990/.local/lib/python2.6/site-packages/saga_python-0.14-py2.6.egg/saga/init.py
```

If saga reports 0.35,  then there must be a mixup at runtime, because that looks like 0.14.  Maybe there is an old version lingering?

**Comment by haoyuanchen** [comment]
I just did "locate saga" and yes, there're lots of old versions existing, including 0.13, 0.14, 0.18...So how can I make sure that everything that uses saga will use the newest version? I can physically remove all the old ones but I guess it's gonna fail to find saga and crash. I'll try anyway.

**Comment by andre-merzky** [comment]
We usually recommend to start with a fresh virtualenv, like this:

```
virtualenv $HOME/ve
source $HOME/ve/bin/activate
pip install --upgrade radical.pilot
```

That basically creates a local python installation in `$HOME/ve` and installs the modules in there.  The `source` line activates that environment --  `which python` should show that it is being used.  If you want to uninstall the modules, you can run:

```
source $HOME/ve/bin/activate
pip uninstall -y radical.pilot
pip uninstall -y saga-python
pip uninstall -y radical.utils
pip install --upgrade radical.pilot
```

updating to a new release would be:

```
source $HOME/ve/bin/activate
pip install --upgrade radical.pilot
```

and at any time you can just remove the virtualenv and recreate it from scratch

As for your current installation: you probably will want to remove all versions for radical.utils, radical.pilot and saga-python, and then do a fresh install of radical.pilot, which will pull the right versions from pypi.

HtH, Andre.

**Comment by haoyuanchen** [comment]
I finally got it to work by also removing all the old saga-python and RP files on Gordon, besides doing what you suggested on my laptop. It seems that even if everything on my laptop is up-to-date, those old files on Gordon will still cause problems. Also, the default python on Gordon seems to be 2.6 and also caused some problems, I aliased it to 2.7 and it works then.

Thanks for your help Andre!
